test: add Jest unit test runner

Robbert opened the pull request.

paulAcato reviewed (commented):
I think we should add the `edit.js(x)` instead of removing it.

In the edit.js you got functions that are not supported in the save.js(x) like memoization (useEffect, useLayoutEffect and useMemo);

Wit more advanced blocks you want to use memoization, i've just debugged for over a week a specific site where this was spamming the editor causing it to run realy slow.

Also in the edit.js u can add stuff like:

```
/**
@param {object} attributes The attributes set in block.json
@param {function} setAttributes The function to update the attributes.
@param {string} clientId The client id of the block which changes when the block is updated.
@param {function} mergeBlocks The function to merge blocks.
@param {function} onReplace The function to replace blocks.
@param {function} onRemove The function to remove blocks.
@param {boolean} isSelected If the block is selected.
@param {string} name The name of the block.
@return {JSX.Element}
@constructor
*/
```

```
import { createBlock } from '@wordpress/blocks';

export default function Edit(
  {
    attributes,
    setAttributes,
    clientId,
    mergeBlocks,
    onReplace,
    onRemove,
    isSelected,
    name
  }
) {
```

```
<RichText
          tagName="p"
          ref={textRef}
          value={attributes.content || removeWrapperFromString( attributes.text ) || ''}
          allowedFormats={allowedFormats}
          onChange={ ( content ) => {
            if (!!content && !!content.includes( '<meta charset="utf-8">' )) {
              content = content.replace( /<meta charSet="utf-8">/gmi, '' );
            }

            setAttributes( {content} )
          } }
          placeholder={
            attributes?.textPlaceholder.replace( /<[^>]*>?/gm, '' ) ||
            __( 'Start typing…', 'nl-design-system-editor' )
          }
          onSplit={ ( content, isOriginal ) => {
            if( ! content && !! isOriginal ) {
              // Keep focus on the current block.
              textRef?.current.focus();

              return content
            }

            const block = createBlock( name, {
              ...attributes,
              content,
            } );

            if ( isOriginal ) {
              block.clientId = clientId;
            }

            return block;
          } }
          onMerge={ mergeBlocks }
          onReplace={ onReplace }
          onRemove={ onRemove }
        />
```

Allowing u to re-add a block in the editor after pressing enter :)

Robbert commented:
@paulAcato I don't think I understand your remark about removing "edit.js" — I didn't remove the file, I just renamed `edit.js` ➝ `edit.jsx` to make it explicit which files need JSX syntax.

After renaming it I refactored it to use `<Paragraph>` from Utrecht instead of maintaining the class names in here too.

If you still have this concern, could you elaborate?

Robbert commented:
I have rebased this PR on latest dependencies.

matijs commented on babel.config.cjs:
```diff
@@ -0,0 +1,10 @@
+/* eslint-env node */
+
+/**
+ * I couldn't refactor this to `.mjs` yet, because I got the following error when running `jest`:
```
`babel.config.json` should work.
